lib/nameMatcher.js

```diff
@@ -0,0 +1,307 @@
+    return _.each(inputCollection, function parseMetaphone(metaphone, name) {
+        inputCollection[name] = metaphone.split(':');
+        return inputCollection;
+    }, {});
+};
+
+
+/**
+ * refresh the firstNames and lastNames by reading from the persistent source (redis db)
+ * @return {Promise} a promise to be fulfilled with bool=true when refresh is complete
+ */
+var refresh = function () {
+    console.log('nameMatcher refreshing local names');
+    var firstNamePromise = promisify(redisClient.hgetall, ['firstNames'], redisClient);
+    var lastNamePromise = promisify(redisClient.hgetall, ['lastNames'], redisClient);
+    var firstNamePromiseDev = promisify(redisClient.hgetall, ['firstNamesDev'], redisClient);
+    var lastNamePromiseDev = promisify(redisClient.hgetall, ['lastNamesDev'], redisClient);
+    return Promise.all(
+        [
+        lastNamePromise,
+        firstNamePromise,
+        lastNamePromiseDev,
+        firstNamePromiseDev
+        ])
+        .then(function storeNamesLocally(names) {
+            // names = [lastNames, firstNames]
+            // firstNames = {bob: 'MP1:MP2', rachel: 'MP3:MP4'}
+            // assign names to local private variables
+            lastNames = parseMetaphones(names[0] || {});
+            firstNames = parseMetaphones(names[1] || {});
+            lastNamesDev = parseMetaphones(names[3] || {});
+            firstNamesDev = parseMetaphones(names[4] || {});
+
+            console.log('nameMatcher refreshing names complete');
+            return true;
+        }).catch(function catchRefreshError(err) {
+            console.error('Encountered error refreshing local names');
+            console.error(err);
+            throw err;
+        });
+};
+
+
+/**
+ * Initialization function. Sets redisClient, and creates whenReady promise that will resolve when the redis connection is ready and data has been populated from the redis DB;
+ * @return {object} nameMatcher
+ */
+var init = function () {
+    console.log('initializing nameMatcher');
+    nameMatcher.whenReady = new Promise(function promisifyWhenReady(resolve, reject) {
+        redisClient = redis.createClient();
+        redisClient.on('ready', function refreshWhenConnectionReady() {
+            console.log('nameMatcher redis connection established');
+            return refresh().then(function resolveWhenReadyPromise() {
+                console.log('nameMatcher initialization complete');
+                resolve(nameMatcher);
+            });
+        });
+    });
+    return nameMatcher;
+};
+
+
+// public methods
+
+/**
+ * Add a new name to the list of names (adds to in-memory list and to persistent storage)
+ * @param first {string} the first name
+ * @param last {string} the last name
+ * @return {Promise} a promise to be fulfilled with a boolean if both names were added (or are already existing)
+ */
+nameMatcher.addName = function (first, last, environment) {
+    if (environment === "prod") {
```

Comment:
After some discussion on the actual deployment, I've opted to put the env in a config file and nix the routes in favor for starting the service on a diff port vs prod/dev